### PR TITLE
fs: Call RELEASE_SEARCH to free resource in pseudorename

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
to avoid the memory leak

### CODE REVIEW

File: fs/vfs/fs_stat.c [hunk 1]
```diff
@@ -58,7 +58,8 @@
  ****************************************************************************/
 
 static inline int statroot(FAR struct stat *buf);
-int stat_recursive(FAR const char *path, FAR struct stat *buf, int resolve);
+static int
+stat_recursive(FAR const char *path, FAR struct stat *buf, int resolve);
```
Comment (btashton) [inline fs/vfs/fs_stat.c:62]: while nxstyle will pass on this I believe the preferred way to handle this is to keep the type on the same line as the function name and then drop the last argument onto the next line aligned with the first argument.

File: fs/vfs/fs_stat.c [hunk 2]
```diff
@@ -93,7 +94,8 @@ static inline int statroot(FAR struct stat *buf)
  *
  ****************************************************************************/
 
-int stat_recursive(FAR const char *path, FAR struct stat *buf, int resolve)
+static int
+stat_recursive(FAR const char *path, FAR struct stat *buf, int resolve)
```
Comment (btashton) [inline fs/vfs/fs_stat.c:98]: Same

Comment (xiaoxiang781216) [reply fs/vfs/fs_stat.c]: Done.

File: fs/vfs/fs_stat.c (hunk 1 above)
Comment (xiaoxiang781216) [reply fs/vfs/fs_stat.c]: Done.

**Review by btashton — APPROVED** [review]
LGTM

**Comment by jerpelea** [comment]
LGTM

**Pull request merged**